Mathlib/Topology/Category/TopCat/Yoneda.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Limits.Preserves.Opposites
+import Mathlib.CategoryTheory.Limits.Preserves.Shapes.Products
+import Mathlib.CategoryTheory.Limits.Shapes.Types
+import Mathlib.Topology.Category.TopCat.Limits.Products
+
+/-!
+
+# Yoneda presheaves on topologically concrete categories
+
+This file develops some API for "topologically concrete" categories, defining universe polymorphic
+"Yoneda presheaves" on such categories.
+-/
+
+universe w w' v u
+
+open CategoryTheory Opposite Limits
+
+variable {C : Type u} [Category.{v} C] (F : C ⥤ TopCat.{w}) (Y : Type w') [TopologicalSpace Y]
+
+namespace ContinuousMap
+
+/--
+A universe polymorphic "Yoneda presheaf" on `C` given by continuous maps into a topoological space
+`Y`.
+-/
+@[simps]
+def yonedaPresheaf : Cᵒᵖ ⥤ Type (max w w') where
```

Comment:
This seems very close to an example of [CategoryTheory.ColimitAdj.restrictedYoneda](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Limits/Presheaf.html#CategoryTheory.ColimitAdj.restrictedYoneda) except that it has a `smallCategory` requirement that can probably be lifted (at least from that definition).

Reply:
To use something like that it seems that I would need `Y` to be an object of a category, which would be `TopCat.{w'}`, and the functor `A` from `restrictedYoneda` would be some sort of topological `ULift`, which we don't have. Or do you have something else in mind?